src/backend/xilinx/memory_axi.rs

```diff
@@ -293,18 +306,23 @@ fn module_mode_fsm(module: &mut v::Module) -> fsm::LinearFsm {
 }
 
 fn bram_logic(
+    name: &str, //assumed to be of form [Memory_controller_axi_<suffix>]
     axi4: &AxiInterface,
     module: &mut v::Module,
     mode_fsm: &fsm::LinearFsm,
     txn_count: v::Expr,
+    data_width: u64,
+    addr_width: u64,
 ) {
-    module.add_decl(v::Decl::new_wire("bram_addr", 5));
-    module.add_decl(v::Decl::new_wire("bram_write_data", 32));
+    module.add_decl(v::Decl::new_wire("bram_addr", addr_width));
+    module.add_decl(v::Decl::new_wire("bram_write_data", data_width));
     module.add_decl(v::Decl::new_wire("bram_we", 1));
-    module.add_decl(v::Decl::new_wire("bram_read_data", 32));
+    module.add_decl(v::Decl::new_wire("bram_read_data", data_width));
     module.add_decl(v::Decl::new_wire("bram_done", 1));
-
-    let mut ram_instance = v::Instance::new("bram", "SINGLE_PORT_BRAM");
+    let suffix_idx = "Memory_controller_axi_".len();
+    let suffix = &name[suffix_idx..];
```

Comment:
Not a big deal, but it could be ever so slightly nicer to take `suffix` in specifically as a parameter?